### ISSUE TITLE
Make the navigate regions sidebars show full height

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#48186]
### Description

When navigating through the editor regions with the Control + backtick keyboard shortcut and any right sidebar is closed, the sidebar region shows only a button to open the related sidebar. For example, in the post editor:

<img width="1787" alt="Screenshot 2023-02-17 at 11 26 43" src="https://user-images.githubusercontent.com/1682452/219619680-d0f97247-64b0-421f-9f73-ae353c99c2cd.png">

In https://github.com/WordPress/gutenberg/pull/48102 @youknowriad made a good point that these sidebars could always be shown full height. For example:

<img width="1792" alt="Screenshot 2023-02-17 at 11 28 24" src="https://user-images.githubusercontent.com/1682452/219619908-4efb7061-3a50-4531-9693-54f8996a7b4f.png">

Making them full height would bring in some pros:
- The 'Open' buttons would be more prominent. Showing something at the top right is better than at the bottom right because of the 'F pattern'. Improved accessibility.
- It would allow some code simplification.

If that's OK from a dsign perspective, I'd be in favor of this change.

The change should be made both in the Post editor and in the Site editor for the sidebars that show the buttons:
- Open document settings
- Open publish panel
- Open save panel

To make this change:
- The changes from https://github.com/WordPress/gutenberg/pull/48102 should be reverted.
- The CSS should be simplified and always set the `top` position to `0`.

### Step-by-step reproduction instructions

- Go to the Post editor.
- Close all sidebars.
- Navigate through the regions by using the Control + backtick keyboard shortcut (or Control+Option+N).
- Observe the regions with the 'Open' buttons show at the bottom right of the screen.
- Go to the Site editor.
- Close all sidebars.
- Navigate through the regions.
- Observe the region with the 'Open' button shows at the bottom right of the screen.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
